### PR TITLE
Fixing unbalanced parentheses in ToString for Post*Assign nodes

### PR DESCRIPTION
**Pull request by bartdesmet**
This fixes issue #11421 which was found during code coverage improvement work.

### CODE REVIEW

File: src/System.Linq.Expressions/src/System/Linq/Expressions/ExpressionStringBuilder.cs [hunk 1]
```diff
@@ -646,6 +646,8 @@ protected internal override Expression VisitUnary(UnaryExpression node)
                     Out("+");
                     break;
                 case ExpressionType.Quote:
+                case ExpressionType.PostIncrementAssign:
+                case ExpressionType.PostDecrementAssign:
```
Comment (bartdesmet) [inline src/System.Linq.Expressions/src/System/Linq/Expressions/ExpressionStringBuilder.cs]: Note I'd likely be better to revisit this method a bit further, and eliminate the `default` case at the bottom by handling all cases explicitly. Just like the printing for binary expressions, the `default` case can then throw an exception for an unexpected node.

Comment (VSadov) [inline src/System.Linq.Expressions/src/System/Linq/Expressions/ExpressionStringBuilder.cs]: Right. Ensuring that we are handling all the cases would be good here.
The original code was likely not caring much about perf and was written at the time the nodes were added/removed. Now it is a stable set and a comprehensive switch would make more sense.

**Comment by VSadov** [comment]
LGTM

**Comment by VSadov** [comment]
There are some merge conflicts

**Comment by bartdesmet** [comment]
After resolving the merge, only the test cases remain. https://github.com/dotnet/corefx/commit/80e15a0f1ea94e4d58bc98da8dc4bfb4f9973062 included the change to add the two cases in the switch table.

**Comment by stephentoub** [comment]
LGTM

**Pull request merged**